### ISSUE TITLE
Budo crashes sometimes

### ISSUE THREAD

**Issue by mattdesl** [mattdesl/budo#107]
With `babelify`, `envify`, `glslify` sometimes the server will crash during dev, without any errors. Difficult to reproduce & debug, only seeing it on one project.

**Comment by mikkoh** [comment]
Here's a console logs of it happening:

<img width="368" alt="screen shot 2015-11-20 at 1 38 27 pm" src="https://cloud.githubusercontent.com/assets/496903/11308412/24a29d20-8f8c-11e5-8e15-cffff8cc2e9c.png">

<img width="465" alt="screen shot 2015-11-17 at 12 23 39 am" src="https://cloud.githubusercontent.com/assets/496903/11308434/494f8930-8f8c-11e5-9b91-d78893a6d465.png">

**Comment by hughsk** [comment]
Yeah, have run into this today too. Can reproduce it by reloading a page which requests a lot of assets — in my case, ~50 — and then reloading frequently. Crashes happen pretty quickly in that case. Going to poke around and see if I can find a fix :)

**Comment by mattdesl** [comment]
Try turning off live reload? Might narrow things down

**Comment by hughsk** [comment]
Yep! Still occurs when I tried (separately):

* Disabling transforms.
* Disabling live reload.
* Switching out `ecstatic` for `st`.

When live reload was disabled, I could see the bundle times coming through the log clearly and they would start to queue up from 100ms to 6000ms if I saved repeatedly. I only get a crash though when I refresh the page too frequently, so I don't think bundling's the issue either...

**Comment by hughsk** [comment]
So, I started logging a running total of the number of requests as they came in, and noticed that a crash occurs at around the same number of requests each time (changing the number of assets loaded per page would change that number)... In fact, the rate of crash seemed to line up super neatly with the amount of logging, as opposed to the number of requests.

Started to suspect that this isn't an issue with budo at all but the wrapper script that's spawning up the process — sure enough, I changed the `child_process.exec` call to `child_process.spawn` and the problem went away :sparkles:

In our project we were streaming the output of `exec` straight to stdout/stderr, but it must still buffer stuff internally. There's a `maxBuffer` option that defaults to 200kB, so it was probably killing the process when it hit that threshold. Unsure if it's the same bug in your case but hope this helps :)
